### PR TITLE
feat(uniqueItems): array unique items

### PR DESCRIPTION
**Pull request by baudicj**
implement uniqueItems : https://swagger.io/docs/specification/data-models/data-types/

### CODE REVIEW

File: lib/modelValidator.js [hunk 1]
```diff
@@ -387,6 +396,23 @@ function validateArray(name, value, field, models, allowBlankTargets, disallowEx
         }
         arrayType = fieldType;
     } else if(models && field.items && field.items.$ref) {
+        if (field.type && field.uniqueItems) {
+            var duplicatedObjects = []
+            for (var i = 0; i < value.length; i++) {
+                var j = i + 1
+                while(j < value.length && !doesArrayContainsValue(duplicatedObjects, value[j])) {
+                    if (_.isEqual(value[i], value[j])) {
+                        duplicatedObjects.push(value[i])
```
Comment (brodyd) [inline lib/modelValidator.js]: Can we break out of both of these loops once we find a duplicate.  We only need to find one after all and then we don't need the duplicate check for the j loop and we can break the i loop too.

File: lib/modelValidator.js [hunk 2]
```diff
@@ -387,6 +396,23 @@ function validateArray(name, value, field, models, allowBlankTargets, disallowEx
         }
         arrayType = fieldType;
     } else if(models && field.items && field.items.$ref) {
+        if (field.type && field.uniqueItems) {
+            var duplicatedObjects = []
+            for (var i = 0; i < value.length; i++) {
+                var j = i + 1
+                while(j < value.length && !doesArrayContainsValue(duplicatedObjects, value[j])) {
+                    if (_.isEqual(value[i], value[j])) {
```
Comment (brodyd) [inline lib/modelValidator.js]: Any chance we can avoid bringing in lodash?  can we do an equality check ourselves?

Comment (brodyd) [reply lib/modelValidator.js]: Perhaps we can use lodash.isEqual if thats all the lodash we need?

File: lib/modelValidator.js (hunk 1 above)
Comment (baudicj) [reply lib/modelValidator.js]: yes, I'll change it

File: lib/modelValidator.js (hunk 2 above)
Comment (baudicj) [reply lib/modelValidator.js]: two soultions :
- create function to compare object (both keys and values)
- just import lodash.isEqual which avoid to import the full lib

File: lib/modelValidator.js (hunk 1 above)
Comment (baudicj) [reply lib/modelValidator.js]: fixed

File: lib/modelValidator.js (hunk 2 above)
Comment (brodyd) [reply lib/modelValidator.js]: I'd prefer we do the compare outselves but I'd accept the lodash.isEqual solution as well.

File: lib/modelValidator.js (hunk 1 above)
Comment (brodyd) [reply lib/modelValidator.js]: Looks good.

File: lib/modelValidator.js (hunk 2 above)
Comment (brodyd) [reply lib/modelValidator.js]: Lodash.isEqual is probably the better solution - even though it means we have a dependency on the lodash.isEqual package.  I was hoping to have no dependencies if possible but comparing objects is kind of nasty so using a know good library is the best solution.  Please change lodash to lodash.isEqual and we can merge this.

Comment (baudicj) [reply lib/modelValidator.js]: done

Comment (brodyd) [reply lib/modelValidator.js]: Thanks

**Pull request merged**